Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs

```diff
@@ -302,12 +367,44 @@ protected APIGatewayProxyResponse MarshallResponse(IHttpResponseFeature response
                     {
                         response.Headers[kvp.Key] = string.Join(",", kvp.Value);
                     }
+
+                    // Remember the Content-Type for possible later use
+                    if (kvp.Key.Equals("Content-Type", StringComparison.CurrentCultureIgnoreCase))
+                        contentType = response.Headers[kvp.Key];
                 }
             }
 
             if (responseFeatures.Body != null)
             {
-                if (responseFeatures.Body is MemoryStream)
+                // Figure out how we should treat the response content
+                var rcEncoding = DefaultResponseContentEncoding;
+                if (contentType != null
+                        && _responseContentEncodingForContentType.ContainsKey(contentType))
+                {
+                    rcEncoding = _responseContentEncodingForContentType[contentType];
+                }
+
+                // Do we encode the response content in Base64 or treat it as UTF-8
+                if (rcEncoding == ResponseContentEncoding.Base64)
+                {
+                    // We want to read the response content "raw" and then Base64 encode it
+                    byte[] bodyBytes;
+                    if (responseFeatures.Body is MemoryStream)
+                    {
+                        bodyBytes = ((MemoryStream)responseFeatures.Body).ToArray();
+                    }
+                    else
+                    {
+                        using (var ms = new MemoryStream())
+                        {
+                            responseFeatures.Body.CopyTo(ms);
+                            bodyBytes = ms.ToArray();
```

Comment:
Binary responses (images etc) can ofter be >80KB. Should we be worried about LOH fragementation in the event of the function being re-used? If so, there's a netstandard version of this package https://github.com/Microsoft/Microsoft.IO.RecyclableMemoryStream coming (or just copy the files) that might help.